src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js

```diff
@@ -840,4 +842,151 @@ describe('ComputeModal', () => {
+        ...defaultAjaxImpl,
+        Runtimes: {
+          runtime: runtimeFunc
+        }
+      }))
+
+      // Act
+      await act(async () => {
+        await render(h(ComputeModalBase, { ...defaultModalProps, tool: runtimeTool.label }))
+
+        await userEvent.click(screen.getByText('Customize'))
+        const selectMenu = await screen.getByLabelText('Application configuration')
+        await userEvent.click(selectMenu)
+        const selectOption = await screen.findByText('Custom Environment')
+        await userEvent.click(selectOption)
+
+        await screen.findByText('Creation Timeout Limit')
+        const timeoutInput = await screen.getByLabelText('Creation Timeout Limit')
+
+        const imageInput = await screen.getByLabelText('Container image')
+        expect(imageInput).toBeInTheDocument()
+        const customImageUri = 'us'
+        await fireEvent.change(imageInput, { target: { value: customImageUri } })
+
+        await fireEvent.change(timeoutInput, { target: { value: 20 } })
+        await userEvent.click(selectMenu)
+      })
+
+      // Act
+      await act(async () => {
+        const nextButton = await screen.findByText('Next')
+        verifyEnabled(nextButton)
+        await userEvent.click(nextButton)
+        const unverifiedDockerWarningHeader = await screen.findByText('Unverified Docker image')
+
+        expect(unverifiedDockerWarningHeader).toBeInTheDocument()
+        const createButton = await screen.findByText('Create')
+        await userEvent.click(createButton)
+      })
+
+      // Assert
+      expect(createFunc).toHaveBeenCalledWith(expect.objectContaining({
+        timeoutInMinutes: 20
+      }))
+    })
+  })
+
+  it.each([
+    { runtimeTool: runtimeTools.Jupyter, imageLabel: defaultImage.label },
+    { runtimeTool: runtimeTools.RStudio, imageLabel: defaultRImage.label }
+  ])('sends null timeout in minutes for tool $runtimeTool.label', async ({ runtimeTool, imageLabel }) => {
+    await act(async () => {
+      // Arrange
+      const createFunc = jest.fn()
+      const runtimeFunc = jest.fn(() => ({
+        create: createFunc,
+        details: jest.fn()
+      }))
+      Ajax.mockImplementation(() => ({
+        ...defaultAjaxImpl,
+        Runtimes: {
+          runtime: runtimeFunc
+        }
+      }))
+
+      // Act
+      await act(async () => {
+        await render(h(ComputeModalBase, { ...defaultModalProps, tool: runtimeTool.label }))
+
+        await userEvent.click(screen.getByText('Customize'))
+        const selectMenu = await screen.getByLabelText('Application configuration')
+        await userEvent.click(selectMenu)
+        const selectOption = await screen.findByText('Custom Environment')
+        await userEvent.click(selectOption)
+
+        await screen.findByText('Creation Timeout Limit')
+        const timeoutInput = await screen.getByLabelText('Creation Timeout Limit')
+        await fireEvent.change(timeoutInput, { target: { value: 20 } })
```

Comment:
Doesn't this test need to be changed to send no value, since this test case is meant to check null timeout submission?

Reply:
Maybe I should leave a comment here, add a new test, or rename this test.

What this line and selectOption2 is verifying that when the image is set to one of our "supported" images, the timeoutInMinutes will be null. 

See: 
https://github.com/DataBiosphere/terra-ui/pull/3560/files/824fb7b477f3f3ac21b1dab8157ea55201ef18b8#diff-a15b61bc748955fee99cca754569a25705e76701992dfc9da20bafbd7dffb664R1580

Reply:
Look it over again and let me know if it clearer!

Reply:
OH yes this is clear! Thanks!